### ISSUE TITLE
[OpenType] Language System Tags for languages using Ɓ with Ƃ-shape

### ISSUE THREAD

**Issue by moyogo** [MicrosoftDocs/typography-issues#1102]
Some Western African orthographies currently use and Southern African orthographies have used Ɓ with a Ƃ-shape as shown in the 1928 International Africa Alphabet of the International Institute of African Languages and Cultures. Most other languages using Ɓ use the B-shape with a top left took like Hausa has since it adopted the letter in its 1928 orthography.
See [Lorna A. Priest, _Request to document glyph variants_, L2/08-034R), 2008-03-18](https://www.unicode.org/L2/L2008/08034r-var-glyphs.pdf).

For example, the following languages  (in some cases some documents have been digitized with the character ɓ for the lowercase of the letter and Ƃ for the uppercase of the letter):
- Dan 'DNJ ' [[dnj](https://iso639-3.sil.org/code/dnj)] in Liberia, generally called Gio, for example "Ƃò" in https://www.bible.com/bible/2828/RUT.2.DHB (see also [L2/08-034R](https://www.unicode.org/L2/L2008/08034r-var-glyphs.pdf#page=5))
- Kpelle (Liberia) 'XPE ' [[xpe](https://iso639-3.sil.org/code/xpe)], see [L2/08-034R](https://www.unicode.org/L2/L2008/08034r-var-glyphs.pdf#page=5)
- Loma 'LOM ' [[lom](https://iso639-3.sil.org/code/lom)], see [L2/08-034R](https://www.unicode.org/L2/L2008/08034r-var-glyphs.pdf#page=5)
- Mano (OT Language System missing) [[mev](https://iso639-3.sil.org/code/mev)], for example in "Ƃii" in https://www.bible.com/bible/2831/MAT.1.MNT

It would be useful to have Language System Tags for the languages known to be using the nowadays less common form to apply substitutions to have the Ƃ-shape, for example with the 'locl' feature, when a font’s default Ɓ has the more common B-sape with a top left hook.

Similary, it would be useful to be able to apply the substitution the other way if a font’s default Ɓ has the Ƃ-shape. For example, Bassa [bsq](https://iso639-3.sil.org/code/bsq) also spoken in Libera seems to commonly use the B-shape of the letter.

Some languages using Ɓ with a B-shape already have OT Language System Tags:

Language System | Language System Tag | Corresponding ISO 639 IDs or other information
--|--|--
| Dan | 'DNJ ' | [dnj](https://iso639-3.sil.org/code/dnj)
| Kpelle (Liberia) | 'XPE ' | [xpe](https://iso639-3.sil.org/code/xpe)
| Loma | 'LOM' | [lom](https://iso639-3.sil.org/code/lom)

Note: The Dan language [dnj](https://iso639-3.sil.org/code/dnj) has at least 3 current orthographies, one for Dan (Gio) in Liberia, one for Eastern Dan in Côte d’Ivoire and one for Western Dan in Côte d’Ivoire. Only Dan (Gio) uses ɓ and Ɓ with a Ƃ-shape.

Some languages that have historicaly used letter with a Ƃ-form or one similar to it, different from the current most common one, already have OT Language System Tags:

Language System | Language System Tag | Corresponding ISO 639 IDs or other information
--|--|--
 Shona | 'SNA0' | [sna](https://iso639-3.sil.org/code/sna)
 Xhosa | 'XHS ' | [xho](https://iso639-3.sil.org/code/xho)
 Zulu | 'ZUL ' | [zul](https://iso639-3.sil.org/code/zul)

Many other languages using Ɓ with Ƃ-form already have OT Languages System Tags:

Language System | Language System Tag | Corresponding ISO 639 IDs or other information
--|--|--
|  Balante | 'BLN ' | [bjt](https://iso639-3.sil.org/code/bjt), [ble](https://iso639-3.sil.org/code/ble)
| Comorian | 'SWB ' | [swb](https://iso639-3.sil.org/code/swb)
| Fulah | 'FF ' | [ff](https://iso639-3.sil.org/code/ff)
| Fulfulde, Adamawa | 'FUB ' | [fub](https://iso639-3.sil.org/code/fub)
| Fulfulde, Borgu | 'FUE ' | [fue](https://iso639-3.sil.org/code/fue)
| Fulfulde, Western Niger | 'FUH ' | [fuh](https://iso639-3.sil.org/code/fuh)
| Hausa | 'HA ' | [ha](https://iso639-3.sil.org/code/ha)
| Kpelle | 'KPE ' | [kpe](https://iso639-3.sil.org/code/kpe)
| Kpelle, Guinea | 'GKP ' | [gkp](https://iso639-3.sil.org/code/gkp)
| Mbo | 'MBO ' | [mbo](https://iso639-3.sil.org/code/mbo)
| Futa | 'FUF ' | [fuf](https://iso639-3.sil.org/code/fuf)
| Nigerian Fulfulde | 'FUV ' | [fuv](https://iso639-3.sil.org/code/fuv)
| Serer | 'SRR ' | [srr](https://iso639-3.sil.org/code/srr)
| Tamashek | 'TMH ' | taq, thv, thz, tmh, ttq
| Toma | 'TOD0' | [tod](https://iso639-3.sil.org/code/tod)

## Proposed OT Language System Tags

Note: at this moment only Mano [mev] seems to prefer the Ƃ-shape in the following list, if anything that one should be added.

Language System | Language System Tag | Corresponding ISO 639 IDs or other information
--|--|--
| Agwagwune | 'YAY ' | [yay](https://iso639-3.sil.org/code/yay)
| Bacama | 'BCY ' | [bcy](https://iso639-3.sil.org/code/bcy)
| Baka | 'BKC ' | [bkc](https://iso639-3.sil.org/code/bkc)
| Bali | 'BCN ' | [bcn](https://iso639-3.sil.org/code/bcn)
| Bamun | 'BAX ' | [bax](https://iso639-3.sil.org/code/bax)
| Bana | 'BCW ' | [bcw](https://iso639-3.sil.org/code/bcw)
| Basaa | 'BAS ' | [bas](https://iso639-3.sil.org/code/bas)
| Bassa | 'BSQ ' | [bsq](https://iso639-3.sil.org/code/bsq)
| Bassari | 'BSC ' | [bsc](https://iso639-3.sil.org/code/bsc)
| Bedjond | 'BJV ' | [bjv](https://iso639-3.sil.org/code/bjv)
| Bomu | 'BMQ ' | [bmq](https://iso639-3.sil.org/code/bmq)
| Buamu | 'BOX ' | [box](https://iso639-3.sil.org/code/box)
| Bura-Pabir | 'BWR ' | [bwr](https://iso639-3.sil.org/code/bwr)
| Burak | 'BYS ' | [bys](https://iso639-3.sil.org/code/bys)
| Bushi | 'BUC ' | [buc](https://iso639-3.sil.org/code/buc)
| Cakfem-Mushere | 'CKY ' | [cky](https://iso639-3.sil.org/code/cky)
| Cicipu | 'AWC ' | [awc](https://iso639-3.sil.org/code/awc)
| Cishingini | 'ASG ' | [asg](https://iso639-3.sil.org/code/asg)
| Daba | 'DBQ ' | [dbq](https://iso639-3.sil.org/code/dbq)
| Dangaléat | 'DAA ' | [daa](https://iso639-3.sil.org/code/daa)
| Dghwede | 'DGH ' | [dgh](https://iso639-3.sil.org/code/dgh)
| Dii | 'DUR ' | [dur](https://iso639-3.sil.org/code/dur)
| Doyayo | 'DOW ' | [dow](https://iso639-3.sil.org/code/dow)
| Duala | 'DUA ' | [dua](https://iso639-3.sil.org/code/dua)
| Dyan | 'DYA ' | [dya](https://iso639-3.sil.org/code/dya)
| Engenni | 'ENN ' | [enn](https://iso639-3.sil.org/code/enn)
| Gbari | 'GBY ' | [gby](https://iso639-3.sil.org/code/gby)
| Gbaya-Mbodomo | 'GMM ' | [gmm](https://iso639-3.sil.org/code/gmm)
| Goemai | 'ANK ' | [ank](https://iso639-3.sil.org/code/ank)
| Gor | 'GQR ' | [gqr](https://iso639-3.sil.org/code/gqr)
| Gude | 'GDE ' | [gde](https://iso639-3.sil.org/code/gde)
| Gulay | 'GVL ' | [gvl](https://iso639-3.sil.org/code/gvl)
| Gwak | 'JGK ' | [jgk](https://iso639-3.sil.org/code/jgk)
| Hdi | 'XED ' | [xed](https://iso639-3.sil.org/code/xed)
| Ik | 'IKX ' | [ikx](https://iso639-3.sil.org/code/ikx)
| Kako | 'KKJ ' | [kkj](https://iso639-3.sil.org/code/kkj)
| Kamwe | 'HIG ' | [hig](https://iso639-3.sil.org/code/hig)
| Karang | 'KZR ' | [kzr](https://iso639-3.sil.org/code/kzr)
| Karekare | 'KAI ' | [kai](https://iso639-3.sil.org/code/kai)
| Kenga | 'KYQ ' | [kyq](https://iso639-3.sil.org/code/kyq)
| Kibaku | 'CKL ' | [ckl](https://iso639-3.sil.org/code/ckl)
| Kim | 'KIA ' | [kia](https://iso639-3.sil.org/code/kia)
| Kimré | 'KQP ' | [kqp](https://iso639-3.sil.org/code/kqp)
| Kuo | 'XUO ' | [xuo](https://iso639-3.sil.org/code/xuo)
| Kwasio | 'NMG ' | [nmg](https://iso639-3.sil.org/code/nmg)
| Lamang | 'HIA ' | [hia](https://iso639-3.sil.org/code/hia)
| Lehar | 'CAE ' | [cae](https://iso639-3.sil.org/code/cae)
| Lobala | 'LOQ ' | [loq](https://iso639-3.sil.org/code/loq)
| Lobi | 'LOB ' | [lob](https://iso639-3.sil.org/code/lob)
| Láá Láá Bwamu | 'BWJ ' | [bwj](https://iso639-3.sil.org/code/bwj)
| Maasina Fulfulde | 'FFM ' | [ffm](https://iso639-3.sil.org/code/ffm)
| Mafa | 'MAF ' | [maf](https://iso639-3.sil.org/code/maf)
| Malba Birifor | 'BFO ' | [bfo](https://iso639-3.sil.org/code/bfo)
| Mango | 'MGE ' | [mge](https://iso639-3.sil.org/code/mge)
| Mano | 'MEV ' | [mev](https://iso639-3.sil.org/code/mev)
| Masana | 'MCN ' | [mcn](https://iso639-3.sil.org/code/mcn)
| Mbuko | 'MQB ' | [mqb](https://iso639-3.sil.org/code/mqb)
| Mbula-Bwazza | 'MBU ' | [mbu](https://iso639-3.sil.org/code/mbu)
| Merey | 'MEQ ' | [meq](https://iso639-3.sil.org/code/meq)
| Mumuye | 'MZM ' | [mzm](https://iso639-3.sil.org/code/mzm)
| Mundang | 'MUA ' | [mua](https://iso639-3.sil.org/code/mua)
| Muyang | 'MUY ' | [muy](https://iso639-3.sil.org/code/muy)
| Mwaghavul | 'SUR0' | [sur](https://iso639-3.sil.org/code/sur)
| Ménik | 'TNR ' | [tnr](https://iso639-3.sil.org/code/tnr)
| Ndut | 'NDV ' | [ndv](https://iso639-3.sil.org/code/ndv)
| Ngas | 'ANC ' | [anc](https://iso639-3.sil.org/code/anc)
| Ninzo | 'NIN ' | [nin](https://iso639-3.sil.org/code/nin)
| Noon | 'SNF ' | [snf](https://iso639-3.sil.org/code/snf)
| Northern Dagara | 'DGI ' | [dgi](https://iso639-3.sil.org/code/dgi)
| Pana (Central African Republic) | 'PNZ ' | [pnz](https://iso639-3.sil.org/code/pnz)
| Parkwa | 'PBI ' | [pbi](https://iso639-3.sil.org/code/pbi)
| Pero | 'PIP ' | [pip](https://iso639-3.sil.org/code/pip)
| Phuie | 'PUG ' | [pug](https://iso639-3.sil.org/code/pug)
| Pulaar | 'FUC ' | [fuc](https://iso639-3.sil.org/code/fuc)
| Pyam | 'PYM ' | [pym](https://iso639-3.sil.org/code/pym)
| Ron | 'CLA ' | [cla](https://iso639-3.sil.org/code/cla)
| Saafi-Saafi | 'SAV ' | [sav](https://iso639-3.sil.org/code/sav)
| Sar | 'MWM ' | [mwm](https://iso639-3.sil.org/code/mwm)
| Sissala | 'SLD ' | [sld](https://iso639-3.sil.org/code/sld)
| Sokoro | 'SOK ' | [sok](https://iso639-3.sil.org/code/sok)
| Tal | 'TAL ' | [tal](https://iso639-3.sil.org/code/tal)
| Tangale | 'TAN ' | [tan](https://iso639-3.sil.org/code/tan)
| Tarok | 'YER ' | [yer](https://iso639-3.sil.org/code/yer)
| Tera | 'TTR ' | [ttr](https://iso639-3.sil.org/code/ttr)
| Tikar | 'TIK ' | [tik](https://iso639-3.sil.org/code/tik)
| Tsikimba | 'KDL ' | [kdl](https://iso639-3.sil.org/code/kdl)
| Tsishingini | 'TSW ' | [tsw](https://iso639-3.sil.org/code/tsw)
| Vai | 'VAI ' | [vai](https://iso639-3.sil.org/code/vai)
| Vute | 'VUT ' | [vut](https://iso639-3.sil.org/code/vut)
| Waja | 'WJA ' | [wja](https://iso639-3.sil.org/code/wja)
| Wandala | 'MFI ' | [mfi](https://iso639-3.sil.org/code/mfi)
| Warji | 'WJI ' | [wji](https://iso639-3.sil.org/code/wji)
| Winyé | 'KST ' | [kst](https://iso639-3.sil.org/code/kst)
| Zulgo-Gemzek | 'GND ' | [gnd](https://iso639-3.sil.org/code/gnd)

**Comment by PeterConstable** [comment]
@moyogo You said,
>It would be useful to have Language System Tags for the languages known to be using the nowadays less common form to apply substitutions to have the Ƃ-shape

To be clear, it's current standard practice for those languages to use the less common shape?

>Note: The Dan language [dnj](https://iso639-3.sil.org/code/dnj) has at least 3 current orthographies, one for Dan (Gio) in Liberia, one for Eastern Dan in Côte d’Ivoire and one for Western Dan in Côte d’Ivoire. Only Dan (Gio) uses ɓ and Ɓ with a Ƃ-shape.

If that's the case, it seems like 'DNJ ' might not be useful since it doesn't determine which shape for Ɓ should be used. So, for instance, if a Web page had 

````
<style type="text/css">
p.dan { font-language-override: "DNJ" }
...
````
one font using 'locl' might display text in the way expected for a given audience, but a different font using 'locl' might display the wrong way for that audience. Does there need to be two tags that reflect the distinct typographic conventions for Dan?

> Some languages that have historicaly used letter with a Ƃ-form or one similar to it, different from the current most common one, already have OT Language System Tags...

So, there's historical typographic conventions that differ from current conventions? That seems like it might indicate those existing tags have the same shortcoming I raised above for DNJ.

>**Proposed OT Language System Tags**
> Note: at this moment only Mano [mev] seems to prefer the Ƃ-shape in the following list, if anything that one should be added.

This is a long list, but your note seems to say only one might really be needed? Why might the others be needed?

**Comment by moyogo** [comment]
> To be clear, it's current standard practice for those languages to use the less common shape?

This seems to be the case. The Dan Gio Bible translation is still in progress, the parts that have been translated use the less common shape. The Liberia Kpelle Bible translation was published in 2014. For Loma the 1971 Bible translation used the less common shape as well. I’m not sure when the Mano Bible translation was published.

However, JW.org material in those languages is using the most common form instead from what I gathered.

> If that's the case, it seems like 'DNJ ' might not be useful since it doesn't determine which shape for Ɓ should be used.

The 1982 and 2014 Eastern Dan orthographies  and the Western Dan orthography use bh instead of Dan Gio ɓ. The 1979 Practical orthography of Côte d’Ivoire languages used bh, but the 1996 revision allow ɓ as well. So in Dan itself in Côte d’Ivoire ɓ may only occur in foreign names or words.

> Does there need to be two tags that reflect the distinct typographic conventions for Dan?

There could be but it doesn’t seem to be required as only Dan Gio uses ɓ. The letters shared between the orthographies don’t have different preferred forms.

> So, there's historical typographic conventions that differ from current conventions? That seems like it might indicate those existing tags have the same shortcoming I raised above for DNJ.

The current conventions in those languages do not used ɓ. Shona, Xhosa and Zulu replaced b/ɓ by b/bh (or bh/b). Ɓ was only used from the 1930 until the 1950s but will occur in reproductions of historical documents.

> This is a long list, but your note seems to say only one might really be needed? Why might the others be needed?

Of the languages that uses the less common shape, only Mano ('MEV ' mev) would need to be added.

The others might be needed if a font is designed for the languages using the less common shape but still supports those languages using the more common shape. That’s not a strong argument I suppose.

**Comment by moyogo** [comment]
> So, IIUC, from your list of 94 possible additions, only 'MEV ' is needed at this time?

Only 'MEV ' is needed if we assume the most common glyph is always the default.

**Comment by PeterCon** [comment]
Added 'MEV '. See the [OT 1.9.1 alpha](https://docs.microsoft.com/en-us/typography/opentype/otspec191alpha/ot191alpha) for draft revisions addressing this issue.

**Comment by PeterCon** [comment]
This is fixed in OT 1.9.1. See the [OT 1.9.1 beta](https://docs.microsoft.com/en-us/typography/opentype/otspec191/ot191beta) for draft revisions addressing this issue. Closing.
